sections/shared/Layout/SideNav/SideNav.tsx

```diff
@@ -20,17 +21,25 @@ import { CryptoCurrency, Synths } from 'constants/currency';
 import { SIDE_NAV_WIDTH, zIndex } from 'constants/ui';
 
 import { MENU_LINKS, MIGRATE_MENU_LINKS } from '../constants';
+import SubMenu from './SubMenu';
 import PriceItem from './PriceItem';
 import PeriodBarStats from './PeriodBarStats';
 import BalanceItem from './BalanceItem';
 import CRatioBarStats from './CRatioBarStats';
 
+const getKeyValue = <T extends object, U extends keyof T>(obj: T) => (key: U) => obj[key];
+
 const SideNav: FC = () => {
 	const { t } = useTranslation();
 	const { asPath } = useRouter();
+	const menuLinkItemRefs = useRef({});
```

Comment:
Using refs here to catch the `y` position of every menu item, so we can then pass it to a props and use `top: y px` so menu and sub menu can be aligned